### ISSUE TITLE
Bug: SQLight: Can't flush None value found in collection Opportunity.tags error on POST opportunity with tags

### ISSUE THREAD

**Issue by Alex-Cannon** [CodeForFoco/volunteercore#83]
When I POST /api/opportunities
with this body:
{
	"partner_name": "VOA",
	"name": "Has Tags",
	"tags": ["art", "delivery"]
}

I get this error:
sqlalchemy.orm.exc.FlushError: Can't flush None value found in collection Opportunity.tags // Werkzeug Debugger

Tested with front-end and postman. Also tested by adding tag id's in place of strings. **I believe this is an API issue. @stevenabadie can you confirm?**

**Comment by Alex-Cannon** [comment]
I may also have the wrong data format for the api. If so, we should update our docs to make it clear what each field expects to receive. Also, we may want to handle this error server-side anyway to stop our site from breaking from bad input.

**Comment by stevenabadie** [comment]
@Alex-Cannon that error would be if a given tag cannot be found. The api does expect the exact case string name of the tag which I would guess is what it happening there. If the tag in your database is 'Art' and try to apply 'art' you will see the error. Here is an example of the endpoint working from current master:
```
http POST http://127.0.0.1:5000/api/opportunities "Authorization: Bearer OMS/lumhOsGJEACAGBhM2FKjzJWOw3O8" partner_name=VOA name="Has tags" tags:='["Event Planning", "Writing"]'
HTTP/1.0 201 CREATED
Content-Length: 567
Content-Type: application/json
Date: Wed, 22 May 2019 23:38:00 GMT
Location: http://127.0.0.1:5000/api/opportunities/3235
Server: Werkzeug/0.15.4 Python/3.6.8

{
    "active": null,
    "commitment_length_months": null,
    "description": null,
    "end_date": null,
    "frequency_modifier": null,
    "frequency_unit": null,
    "id": 3235,
    "location_city": null,
    "location_state": null,
    "location_street": null,
    "location_zip": null,
    "name": "Has tags",
    "partner_name": "VOA",
    "partner_string": "VOA",
    "shift_hours": null,
    "start_date": null,
    "tag_count": 2,
    "tags": {
        "Skill": [
            "Event Planning",
            "Writing"
        ]
    },
    "training_time_hours": null,
    "volunteers_needed": null

```

**Comment by stevenabadie** [comment]
I'll add an error in that endpoint for the tags. Do you think it would be better to return bad request or allow to  let the POST complete if at least one tag is good?

**Comment by Alex-Cannon** [comment]
@stevenabadie I think it'd be best to return a 400 bad request because that means the front-end isn't handling tags correctly.

I see why there was an error now. I was trying to post a **tag category**, not a tag. Do you think users should be prompted to choose a tag or a tag category?

**Comment by stevenabadie** [comment]
That goes back to the discussion of having multiple tag categories. In reviewing the endpoint I found another error point. If there are duplicate tag names across two or more tag categories the function will pick the first match which could be the wrong one.

The opportunity endpoint is completely ignoring the tagcategory of the tag right now because I originally built the tag model to require a unique name. I removed the unique constraint recently when I realized the VOA data has duplicates across Skills and Passions.

We could provide a tag form for each tagcategory. Alternatively I can turn the unique constraint back on for tag names and change the name of the duplicate tags in the VOA data. With the latter, the current endpoint works in only requiring the name string of the tag.

Edit: Personally I prefer the latter
